events/events.h

```diff
@@ -85,18 +85,26 @@ int events_timer_reset(void *);
  * associated with expired timers registered via events_timer_register will
  * be run.  If any event function returns a non-zero result, no further
  * events will be run and said non-zero result will be returned; on error,
- * -1 will be returned.
+ * -1 will be returned.  May be interrupted by events_interrupt, in which
+ * case -2 will be returned.
  */
 int events_run(void);
 
 /**
  * events_spin(done):
  * Run events until ${done} is non-zero (and return 0), an error occurs (and
  * return -1), or a callback returns a non-zero status (and return the status
- * code from the callback).
+ * code from the callback).  May be interrupted by events_interrupt, in which
+ * case -2 will be returned.
  */
 int events_spin(int *);
 
+/**
+ * events_interrupt(signal):
+ * Halt the event loop after finishing the current event.
+ */
+void events_interrupt(int);
```

Comment:
The parameter here is weird.  I'd say that it makes sense if you want to register this as a signal handler... but I can't imagine what that would accomplish since if you do that you would exit the event loop and then not know what happened.

But this should be documented as being something which can be called safely from within a signal handler.